File: src/main/java/com/financemanager/controllers/MenuController.java
package com.financemanager.controllers;

import javafx.application.Platform;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Node;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.layout.StackPane;
import javafx.stage.FileChooser;
import javafx.stage.Modality;
import javafx.stage.Stage;

import java.io.File;
import java.io.IOException;

public class MenuController {

    @FXML private Button create_new_manager_button;
    @FXML private Button load_manager_button;
    @FXML private Button manage_categories_button;
    @FXML private Button exit_button;
    @FXML private Button import_manager_button;
    @FXML private Button settings_button;
    @FXML private StackPane rootPane;

    @FXML
    public void initialize() {
        Platform.runLater(() -> {
            Stage stage = (Stage) rootPane.getScene().getWindow();
            stage.setResizable(false);
            stage.sizeToScene();
            stage.setTitle("Finance Manager - Main Menu");
            stage.centerOnScreen();
        });
    }

    @FXML
    public void handleCreateManager(ActionEvent event) {
        try {
            FXMLLoader loader = new FXMLLoader(getClass().getResource("/CreateManager.fxml"));
            Parent root = loader.load();
            Stage stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
            stage.setScene(new Scene(root));
            stage.setTitle("Create Manager");
            stage.show();
        } catch (IOException e) {
            System.out.println("Error loading FXML: " + e.getMessage());
        }
    }

    @FXML
    public void handleLoadManager(ActionEvent event) {
        try {
            FXMLLoader loader = new FXMLLoader(getClass().getResource("/LoadManager.fxml"));
            Parent root = loader.load();
            Stage stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
            stage.setScene(new Scene(root));
            stage.setResizable(false);
            stage.sizeToScene();
            stage.centerOnScreen();
            stage.setTitle("Load Manager");
        } catch (IOException e) {
            System.out.println("Error loading FXML: " + e.getMessage());
        }
    }

    @FXML
    public void handleImportManager(ActionEvent event) {
        FileChooser fileChooser = new FileChooser();
        fileChooser.setTitle("Import Existing Manager");
        fileChooser.getExtensionFilters().addAll(
                new FileChooser.ExtensionFilter("Manager Files (*.manager)", "*.manager"),
                new FileChooser.ExtensionFilter("All Files", "*.*")
        );

        Stage stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
        File selectedFile = fileChooser.showOpenDialog(stage);

        if (selectedFile != null) {
            System.out.println("Selected File: " + selectedFile.getAbsolutePath());
            // TODO: carregar o manager e navegar para o Dashboard
        } else {
            System.out.println("Selection Cancelled");
        }
    }

    @FXML
    public void handleExit(ActionEvent event) {
        Platform.exit();
        System.exit(0); // exitProcess(0) do Kotlin equivale a System.exit(0) em Java
    }

    @FXML
    public void handleSettings(ActionEvent event) {
        try {
            FXMLLoader loader = new FXMLLoader(getClass().getResource("/SettingsView.fxml"));
            Parent root = loader.load();
            Stage settingsStage = new Stage();
            settingsStage.setTitle("Settings");
            settingsStage.setScene(new Scene(root));
            settingsStage.initModality(Modality.APPLICATION_MODAL);
            settingsStage.initOwner(((Node) event.getSource()).getScene().getWindow());
            settingsStage.showAndWait();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @FXML
    public void handleCategories(ActionEvent event) {
        try {
            FXMLLoader loader = new FXMLLoader(getClass().getResource("/CategoriesView.fxml"));
            Parent root = loader.load();
            Stage categoriesStage = new Stage();
            categoriesStage.setTitle("Categories");
            categoriesStage.setScene(new Scene(root));
            categoriesStage.initModality(Modality.APPLICATION_MODAL);
            categoriesStage.initOwner(((Node) event.getSource()).getScene().getWindow());
            categoriesStage.showAndWait();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/com/financemanager/controllers/MenuController.kt
package com.financemanager.controllers

import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.scene.Node
import javafx.stage.Stage
import javafx.event.ActionEvent
import javafx.scene.control.Button
import javafx.application.Platform
import javafx.scene.layout.StackPane
import javafx.stage.FileChooser
import javafx.stage.Modality
import java.io.File

import java.io.IOException
import kotlin.system.exitProcess


class MenuController {

    @FXML var create_new_manager_button: Button = Button()
    @FXML var load_manager_button: Button = Button()
    @FXML var manage_categories_button: Button = Button()
    @FXML var exit_button: Button = Button()
    @FXML var import_manager_button: Button = Button()
    @FXML var settings_button: Button = Button()
    @FXML var rootPane: StackPane = StackPane()

    @FXML
    fun initialize() {
        javafx.application.Platform.runLater {
            val stage = rootPane.scene.window as javafx.stage.Stage
            stage.isResizable = false
            stage.sizeToScene()
            stage.title = "Finance Manager - Main Menu"
            stage.centerOnScreen()
        }
    }

    @FXML
    fun handleCreateManager(event: ActionEvent) {
        try {
            val loader = FXMLLoader(javaClass.getResource("/CreateManager.fxml"))
            val settingsRoot = loader.load<Parent>()

            val menuScene = Scene(settingsRoot)

            val currentStage = (event.source as Node).scene.window as Stage

            currentStage.apply {
                scene = menuScene
                title = "Create Manager"
                show()
            }
        } catch (e: IOException) {
            println("Error loading FXML: ${e.message}")
        }
    }

    @FXML
    fun handleLoadManager(event: ActionEvent) {
        val loader = FXMLLoader(javaClass.getResource("/LoadManager.fxml"))
        val root = loader.load<Parent>()
        val stage = (event.source as Node).scene.window as Stage
        stage.scene = Scene(root)
        stage.isResizable = false
        stage.sizeToScene()
        stage.centerOnScreen()
        stage.title = "Load Manager"
    }

    @FXML
    fun handleImportManager(event: ActionEvent) {
        val fileChooser = FileChooser()
        fileChooser.title = "Import Existing Manager"

        fileChooser.extensionFilters.addAll(
            FileChooser.ExtensionFilter("Manager Files (*.manager)", "*.manager"),
            FileChooser.ExtensionFilter("All Files", "*.*")
        )
        val currentStage = (event.source as Node).scene.window as Stage

        val selectedFile: File? = fileChooser.showOpenDialog(currentStage)

        if (selectedFile != null) {
            println("Selected File: ${selectedFile.absolutePath}")
        } else {
            println("Selection Cancelled")
        }
    }

    @FXML
    fun handleExit(event: ActionEvent) {
        Platform.exit()
        exitProcess(0)
    }

    @FXML
    fun handleSettings(event: ActionEvent) {
        val loader = FXMLLoader(javaClass.getResource("/SettingsView.fxml"))
        val root = loader.load<Parent>()

        val settingsStage = Stage().apply {
            title = "Settings"
            scene = Scene(root)

            initModality(Modality.APPLICATION_MODAL)
            initOwner((event.source as Node).scene.window)
        }
        settingsStage.showAndWait()
    }

    @FXML
    fun handleCategories(event: ActionEvent) {
        val loader = FXMLLoader(javaClass.getResource("/CategoriesView.fxml"))
        val root = loader.load<Parent>()

        val categoriesStage = Stage().apply {
            title = "Categories"
            scene = Scene(root)

            initModality(Modality.APPLICATION_MODAL)
            initOwner((event.source as Node).scene.window)
        }
        categoriesStage.showAndWait()
    }

}